gmso/core/topology.py

```diff
@@ -965,6 +966,171 @@ def get_index(self, member):
 
         return index
 
+    def to_datatables(self, parameter="sites", labels=None, unyts_bool=True):
+        """Return a pandas dataframe object for the sites in a topology
+
+        Parameters
+        ----------
+        parameter : a string
+            A string determining what aspects of the gmso topology will be reported. Options are: 'sites','bonds','angles'
+        and 'dihedrals'. Defaults to 'sites'.
+        labels : List of strings that are attributes of the topology site.
+            Examples of these can be found by printing `dir(topology.sites[0])`
+        unyts_bool: bool
+            Determine if numerical values are saved as unyt quantities or floats. See
+        https://unyt.readthedocs.io/en/stable/usage.html
+        for more information about manipulating unyt quantities.
+        Default is True.
+
+        Returns
+        -------
+        Pandas Dataframe
+            A pandas.Dataframe object, see https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html
+            for further information.
+
+        Examples
+        ________
+        >>> topology.to_dataframes(parameter = 'sites', labels = ['charge'])
+            This will return a dataframe with a listing of the sites and include the charges that correspond to each site.
+        >>> topology.to_dataframes(parameter = 'dihedrals', labels = ['positions'])
+            This will return a dataframe with a listing of the sites that make up each dihedral, the positions of each of
+            those sites, and the parameters that are associated with the dihedrals.
+
+        Notes
+        ____
+        A dataframe is easily manipulated. In order to change the rounding to two decimals places for a column named `label`:
+            >>> df['label'] = df['label'].round(2)
+        The column labels can also be easily modified. This line can take a dataframe `df` and rename a column labeled
+        `Atom0` to `newname` using a dictionary.
+            >>> df.rename(columns = {'Atom0':'newname'})
```

Comment:
we can add a reference to pandas documentation here

Reply:
Done!